### ISSUE TITLE
Make it possible to use the human-readable schema format

### ISSUE THREAD

**Issue by advanderveer** [cedar-policy/vscode-cedar#17]
### Category

Cedar editing features

### Describe the feature you'd like to request

The [human-readable schema format](https://docs.cedarpolicy.com/schema/human-readable-schema-grammar.html) is recommend according to the official documentation: https://docs.cedarpolicy.com/schema/schema.html#schema. 

But I don't see a way for this extension to use it directly.

### Describe alternatives you've considered

Manually converting the human readable format to the json format. But doing this on every change is painful when designing and iterating on the policies. 

I could write the schema in json, but it doesn't support comments. Which makes it hard to reason about in a team

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request
- [ ] ⚠️ This feature might incur a breaking change

**Comment by hakanson** [comment]
I'm working on this now as part of #10.  Would you expect a feature to auto translate to JSON (for use in pre 3.1 versions)?

As a temporary solution, I have defined a `.vscode\tasks.json` using the Cedar CLI that can be manual run from inside VSCode.  I haven't figured on how to trigger a task "on save" for a specific file extension (but also am spending my time on the built-in extension code).  This does require `cedar` to be on your path.

```json
{
  "version": "2.0.0",
  "tasks": [
    {
      "label": "Cedar CLI translate-schema human-to-json",
      "type": "shell",
      "command": "cedar",
      "args": [
        "translate-schema",
        "--schema",
        "${file}",
        "--direction",
        "human-to-json",
        "|",
        "jq",
        ">",
        "${file}.json"
      ]
    }
  ]
}
```

**Comment by advanderveer** [comment]
Thank you for the task idea, that is nice. 

I have two thoughts, not sure how related they are to what you're asking:
- It would probably be nice to allow the user to select which version of Cedar to use for the extension. For example AVP doesn't support 3.0 so users might want to specify a 2.x version in their workspace (maybe just by specifying the wasm/binary used for cedar)
- For how I'm using it, it would just be valuable to have validation/and code suggestions be based on the human-readable format during development. I guess that doesn't require compilation to the JSON format persee, but it could be useful for other tools to read.

**Comment by hakanson** [comment]
pre-release v0.7.2 added support to translate between JSON and human-readable schemas. along with some basic syntax highlighting and navigation.  Either name the file `cedarschema` or `*.cedarschema` or **Set Language Mode** to `Cedar Schema`

However, IntelliSense (auto-complete) requires the JSON-based schema (for now).  At the top of the Cedar Schema file is a clickable Code Lens line that will run translation (also available in context menu and as `Cedar: Translate Cedar schema` in command palette.

![Translate Cedar Schema](https://github.com/cedar-policy/vscode-cedar/assets/414500/abaea1ae-0fc4-42b6-b61a-16443a1d5d4b)

**Comment by cdisselkoen** [comment]
I assume this can be closed based on #21?

**Comment by hakanson** [comment]
Closing this since pre-release v0.7.3 supports this.  Cedar 2.x vs Cedar 3.x is part of #10
